Makefile

```diff
@@ -161,7 +161,7 @@ reviewable: generate lint
 # Ensure branch is clean.
 check-diff: reviewable
 	@$(INFO) checking that branch is clean
-	@if [ $$(git status -s | wc -c) != "0" ]; then $(FAIL); fi;
+	@if [ $$(git status -s | wc -c | sed -e 's/^[ \t]*//') != "0" ]; then $(FAIL); fi;
```

Comment:
This is getting a little arcane. Could we take advantage of `git status --porcelain`, which is intended to be parsed by scripts?

```shell
$ git status
On branch rc
nothing to commit, working tree clean
$ git status --porcelain
$ touch moo
$ git status --porcelain
?? moo
$ echo HI > README.md 
$ git status --porcelain
 M README.md
?? moo
$ test -z "$(git status --porcelain)" && echo clean
$ git reset --hard
HEAD is now at 8ab0cb15 Final proof-reading pass
$ rmd moo
$ test -z "$(git status --porcelain)" && echo clean
clean
```

Reply:
Using `--porcelain` seems fine. `git status -s` is [almost the same thing](https://www.stefanjudis.com/today-i-learned/the-short-version-of-git-status-and-the-close-but-different-porcelain-mode/), but definitely think using `test` is a big improvement.